client/components/jetpack/upsell-product-wpcom-plan-card/index.tsx

```diff
@@ -0,0 +1,200 @@
+import {
+	TERM_ANNUALLY,
+	getPlan,
+	getPlanFeaturesObject,
+	Plan,
+	PlanSlug,
+	isJetpackBackupSlug,
+	isJetpackScanSlug,
+	isJetpackSearchSlug,
+} from '@automattic/calypso-products';
+import { Gridicon } from '@automattic/components';
+import { formatCurrency } from '@automattic/format-currency';
+import { useTranslate } from 'i18n-calypso';
+import { useMemo } from 'react';
+import BackupImage from 'calypso/assets/images/jetpack/rna-image-backup.png';
+import DefaultImage from 'calypso/assets/images/jetpack/rna-image-default.png';
+import ScanImage from 'calypso/assets/images/jetpack/rna-image-scan.png';
+import SearchImage from 'calypso/assets/images/jetpack/rna-image-search.png';
+import DisplayPrice from 'calypso/components/jetpack/card/jetpack-product-card/display-price';
+import JetpackRnaActionCard from 'calypso/components/jetpack/card/jetpack-rna-action-card';
+import slugToSelectorProduct from 'calypso/my-sites/plans/jetpack-plans/slug-to-selector-product';
+import { useSelector } from 'calypso/state';
+import { getCurrentUserCurrencyCode } from 'calypso/state/currency-code/selectors';
+import { getSiteAvailableProduct } from 'calypso/state/sites/products/selectors';
+import { getSelectedSiteSlug } from 'calypso/state/ui/selectors';
+import type { Duration, SelectorProduct } from 'calypso/my-sites/plans/jetpack-plans/types';
+
+import './style.scss';
+
+interface UpsellProductWpcomPlanCardProps {
+	nonManageProductSlug: string;
+	WPcomPlanSlug: PlanSlug;
+	siteId: number | null;
+	onCtaButtonClick: () => void;
+}
+
+const UpsellProductWpcomPlanCard: React.FC< UpsellProductWpcomPlanCardProps > = ( {
+	nonManageProductSlug,
+	WPcomPlanSlug,
+	siteId,
+	onCtaButtonClick,
+} ) => {
+	const translate = useTranslate();
+	const currencyCode = useSelector( getCurrentUserCurrencyCode );
+	const selectedSiteSlug = useSelector( getSelectedSiteSlug );
+	const item = slugToSelectorProduct( nonManageProductSlug ) as SelectorProduct;
+	const plan = getPlan( WPcomPlanSlug ) as Plan;
+	const planSlug = plan.getStoreSlug();
+	const jetpackFeatures = plan?.get2023PricingGridSignupJetpackFeatures?.();
+	const jetpackFeaturesObject = getPlanFeaturesObject( jetpackFeatures );
+	const siteProduct = useSelector( ( state ) =>
+		getSiteAvailableProduct( state, siteId, planSlug )
+	);
+
+	const billingTerm: Duration = TERM_ANNUALLY;
+	const ctaButtonURL: string = `https://wordpress.com/checkout/${ selectedSiteSlug }/${ planSlug }?redirect_to=${ encodeURIComponent(
+		window.location.href
+	) }`;
+	const isFetchingPrices: boolean = ! siteProduct;
+	const originalPrice: number = siteProduct?.cost_smallest_unit ?? 0;
+	const displayPrice: number = siteProduct?.cost / 12 ?? 0;
```

Comment:
Curious what the reason is for using `cost_smallest_unit` and `cost`?

Confidence check: We're not using `combined_cost_display` and `cost_per_month_display` because they already have the currency embedded, which won't work for the Upsell layout.
![image](https://github.com/Automattic/wp-calypso/assets/6586048/7c261e29-946b-48b2-a5be-7d0292f3e9e5)